subresource-loading.bs

```diff
@@ -59,41 +60,286 @@ Note: This specification is under construction. See
+   from the document</a> during the previous step.
+
+   Note: This is specified consistently with
+   <a href="https://github.com/whatwg/html/pull/2673">whatwg/html#2673</a>.
+   Currently we don't fire `error` events in this case. If we change the
+   decision at
+   <a href="https://github.com/whatwg/html/pull/2673">whatwg/html#2673</a> to
+   fire `error` events, then change this step accordingly.
+
+1. Assert: |element|'s <a spec="html">node document</a> is equal to |element|'s
+   <a href="spec">preparation-time document</a>.
+1. If |result|'s [=web bundle result/error to rethrow=] is not null, then:
+
+   1. <a spec="html">Report the exception</a> given |result|'s [=web bundle
+      result/error to rethrow=].
+
+      ISSUE: There are no relevant
+      <a href="https://html.spec.whatwg.org/multipage/webappapis.html#concept-script">script</a>,
+      because [=web bundle result=] isn't a
+      <a href="https://html.spec.whatwg.org/multipage/webappapis.html#concept-script">script</a>.
+      This needs to wait for
+      <a href="https://github.com/whatwg/html/issues/958">whatwg/html#958</a>
+      before it is fixable.
+
+   1. Return.
+
+1. Assert: |result|'s [=web bundle result/registration=] is not null.
+1. If |result|'s [=web bundle result/registration=]'s [=web bundle
+   registration/fetch entry=]'s [=web bundle fetch entry/state=] is "failed":
+   1. [=Fire an event=] named `error` at |element|.
+   1. Return.
+1. Assert: |result|'s [=web bundle result/registration=]'s [=web bundle
+   registration/fetch entry=]'s [=web bundle fetch entry/state=] is "fetched".
+1. [=Fire an event=] named `load` at |element|.
+
+## Removing ## {#removing}
+
+If `script` element is
+<a spec="html" href="https://html.spec.whatwg.org/multipage/infrastructure.html#remove-an-element-from-a-document">removed
+from the document</a>, user agents must run the following algorithm:
+
+1. If <a spec="html">the script's type</a> is not "`webbundle`", then return.
+1. If [=the script's result=] is null, then return.
+1. Assert: [=the script's result=] is an [=web bundle result=].
+1. Let |registration| be [=the script's result=]'s [=web bundle
+   result/registration=].
+1. Set [=the script's result=] to null.
+1. If |registration| is null, then return.
+1. Let |document| be the <a spec="html">node document</a>.
+1. Assert: |document|'s [=Document/web bundle registration list=]
+   [=list/contains=] |registration|.
+1. [=list/Remove=] |registration| from |document|'s [=Document/web bundle
+   registration list=].
+1. [=Queue a microtask=] to perform the following steps:
+
+   1. Let |fetch entry| be |registration|'s [=web bundle registration/fetch
+      entry=].
+   1. If |fetch entry| is [=web bundle fetch entry/used by a registration=] in
+      |document|, then return.
+
+   1. [=list/Remove=] |fetch entry| from |document|'s [=Document/web bundle
+      fetch entry list=].
+
+      Note: It is possible that |document|'s [=Document/web bundle fetch entry
```

Comment:
Is this still possible? Since the step 5.1.2 of **prepare a web bundle** was removed as per [Hayato's comment](https://github.com/WICG/webpackage/pull/721#discussion_r838384360), a fetch entry cannot be shared by multiple registrations in a document.

Reply:
Yes, possible. Consider the script below:

```
const script1 = document.createElement('script');
// Set type=webbundle and textContent.
const script2 = document.createElement('script');
// Set type=webbundle and textContent with the same source URL.

// A fetch entry is added to the web bundle fetch entry list,
document.head.appendChild(script1);
script1.remove();
// The fetch entry is reused here.
document.head.appendChild(script2);
script2.remove();
```

Here we have two pending microtasks that attempt to remove the fetch entry from the web bundle fetch entry list.
The first microtask will actually remove the fetch entry from the list, and when the second microtask is executed the fetch entry is no longer in the list.

Reply:
This depends on how `used by a registration` is defined.
In the current PR, a fetch entry is not considered `used` even if it is referenced from a registration pending removal.
Alternatively we might be able to define it more like reference counting or garbage collecting (i.e. remove the fetch entry from the list when the fetch entry is about to be destructed). In that case, this corner case would disappear. (I'm a little less sure how to spec in this way though)
So far I don't see any specific reasons to prefer each of these ways: while there are very slight observable behavior diferrences, I don't expect the differences have any real impacts for ordinal use cases.

Reply:
Understood. Thanks for the explanation!

I agree that the difference in behavior in this corner case does not have any impact for ordinal use cases. Current definition of `used by a registration` looks good to me.